src/Illuminate/Support/Str.php

```diff
@@ -1219,23 +1217,30 @@ private static function toStringOr($value, $fallback)
      */
     public static function replace($search, $replace, $subject, $caseSensitive = true)
     {
-        if ($search instanceof Traversable) {
-            $search = (new Collection($search))->all();
-        }
-
-        if ($replace instanceof Traversable) {
-            $replace = (new Collection($replace))->all();
-        }
-
-        if ($subject instanceof Traversable) {
-            $subject = (new Collection($subject))->all();
-        }
+        $search = self::toArrayIfTraversable($search);
+        $replace = self::toArrayIfTraversable($replace);
+        $subject = self::toArrayIfTraversable($subject);
 
         return $caseSensitive
             ? str_replace($search, $replace, $subject)
             : str_ireplace($search, $replace, $subject);
     }
 
+    /**
+     * Convert a Traversable to an array, or return the original value if not Traversable.
+     *
+     * @param  mixed  $value
+     * @return mixed
```

Comment:
You can probably enhance is by using a conditional return value type:
```suggestion
     * @return ($value is \Traversable ? array : mixed)
```